org_fedora_oscap/service/kickstart.py

```diff
@@ -0,0 +1,253 @@
+
+        if not self.profile_id:
+            self.profile_id = "default"
+
+        if self.content_type in ("rpm", "archive") and not self.content_path:
+            msg = "Path to the XCCDF file has to be given if content in RPM "\
+                  "or archive is used"
+            raise KickstartValueError(msg)
+
+        if self.content_type == "rpm" and not self.content_url.endswith(".rpm"):
+            msg = "Content type set to RPM, but the content URL doesn't end "\
+                  "with '.rpm'"
+            raise KickstartValueError(msg)
+
+        if self.content_type == "archive":
+            supported_archive = any(
+                self.content_url.endswith(arch_type)
+                for arch_type in common.SUPPORTED_ARCHIVES
+            )
+            if not supported_archive:
+                msg = "Unsupported archive type of the content "\
+                      "file '%s'" % self.content_url
+                raise KickstartValueError(msg)
+
+        # do some initialization magic in case of SSG
+        if self.content_type == "scap-security-guide":
+            if not common.ssg_available():
+                msg = "SCAP Security Guide not found on the system"
+                raise KickstartValueError(msg)
+
+            self.content_path = common.SSG_DIR + common.SSG_CONTENT
+
+    def __str__(self):
+        """Generate the kickstart representation.
+
+        What should end up in the resulting kickstart file,
+        i.e. string representation of the stored data.
+
+        :return: a string
+        """
+        if not self.profile_id:
+            return ""
+
+        ret = "%%addon %s" % self.name
+        ret += "\n%s" % key_value_pair("content-type", self.content_type)
+
+        if self.content_url:
+            ret += "\n%s" % key_value_pair("content-url", self.content_url)
+
+        if self.datastream_id:
+            ret += "\n%s" % key_value_pair("datastream-id", self.datastream_id)
+
+        if self.xccdf_id:
+            ret += "\n%s" % key_value_pair("xccdf-id", self.xccdf_id)
+
+        if self.content_path and self.content_type != "scap-security-guide":
+            ret += "\n%s" % key_value_pair("content-path", self.content_path)
+
+        if self.cpe_path:
+            ret += "\n%s" % key_value_pair("cpe-path", self.cpe_path)
+
+        if self.tailoring_path:
+            ret += "\n%s" % key_value_pair("tailoring-path", self.tailoring_path)
+
+        ret += "\n%s" % key_value_pair("profile", self.profile_id)
+
+        if self.fingerprint:
+            ret += "\n%s" % key_value_pair("fingerprint", self.fingerprint)
+
+        if self.certificates:
+            ret += "\n%s" % key_value_pair("certificates", self.certificates)
+
+        ret += "\n%end\n\n"
+        return ret
+
+
+class OSCAPKickstartSpecification(KickstartSpecification):
+    """The kickstart specification of the OSCAP service."""
+
+    addons = {
+        "org_fedora_oscap": OSCAPKickstartData
```

Comment:
Btw. this is the place where you can "register" another add-on name. You will have to update the `process_kickstart` and `setup_kickstart` methods to make it work with the rest of the module, but that should be it.